test/jdk/javax/swing/JCheckBox/8032667/bug8032667.java

```diff
@@ -20,55 +20,131 @@
  * or visit www.oracle.com if you need additional information or have any
  * questions.
  */
+
 import java.awt.BorderLayout;
 import java.awt.Canvas;
 import java.awt.Dimension;
-import java.awt.Graphics;
+import java.awt.FlowLayout;
 import java.awt.Graphics2D;
+import java.awt.Graphics;
 import java.awt.Image;
+import java.awt.event.WindowAdapter;
+import java.awt.event.WindowEvent;
 import java.awt.image.BufferedImage;
-import javax.swing.JApplet;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import javax.swing.BoxLayout;
+import javax.swing.JButton;
 import javax.swing.JCheckBox;
 import javax.swing.JComponent;
+import javax.swing.JFrame;
+import javax.swing.JPanel;
+import javax.swing.JTextArea;
+import javax.swing.JTextField;
 import javax.swing.SwingUtilities;
+import javax.swing.text.JTextComponent;
 
 /* @test
  * @bug 8032667
  * @summary [macosx] Components cannot be rendered in HiDPI to BufferedImage
- * @run applet/manual=yesno bug8032667.html
+ * @run main bug8032667
```

Comment:
I suggest do not convert manual tests from the applet to the regular tests, until we will not find a way to unify them, in a similar way as it was for "applet/manual=yesno". Otherwise, all our manual tests will look differently, since different people will make the UI differently.

Reply:
Unifying the UI makes sense in the long run. Will it involve retrofitting all the existing manual test which already have its own UI?

In most cases, the UI in manual tests is similar: Pass / Fail buttons at the bottom with the test instructions above. Then there are differences… Other parts of the UI depend on the nature of the test.

I've always found manual tests a bit confusing: a test, especially applet-based one, opens two windows; however, one window is easier to focus on.

Reply:
To confirm: we're not modifying the current applet-based manual tests until there's a common framework to migrate them off the applet API. Do I understand it correctly?

Reply:
Yes, I suggest that. Maybe we can start with that replacement? In the jtreg itself or via separate lib.

Reply:
A separate lib seems more convenient as it will allow running such tests without test harness which is quicker and useful for debugging.